front/components/assistant_builder/ActionsScreen.tsx

```diff
@@ -578,7 +654,35 @@ function ActionConfigEditor({
   setEdited: (edited: boolean) => void;
   description: string;
   onDescriptionChange: (v: string) => void;
-}) {
+};
+
+function ActionConfigEditor({
+  owner,
+  action,
+  vaultsUsedInActions,
+  instructions,
+  updateAction,
+  setEdited,
+  description,
+  onDescriptionChange,
+}: ActionConfigEditorProps) {
+  const { vaults } = useContext(AssistantBuilderContext);
+
+  // Only allow one vault across all actions
+  const allowedVaults = useMemo(() => {
+    const usedVaultsInOtherActions = vaults.reduce((acc, v) => {
+      return (vaultsUsedInActions[v.sId] ?? []).some((a) => a !== action)
```

Comment:
Can we use the action name instead of the reference?

Reply:
I wanted to but the name is not unique, therefor I can't make sure which one are the "others" later.

Reply:
I guess the name isn't unique in the UI but for pretty sure that when saving it must be unique.